src/klein/_plating.py

```diff
@@ -237,21 +237,15 @@ def mymethod(
                     json_data.update(data)
                     for ignored in self._presentationSlots:
                         json_data.pop(ignored, None)
-                    str = "json"
+                    request.setHeader(b"content-type", b"application/json")
                     ready = yield resolveDeferredObjects(json_data)
                     result = dumps(ready)
                 else:
                     data[self.CONTENT] = loader.load()
-                    str = "html"
+                    request.setHeader(
+                        b"content-type", b"text/html; charset=utf-8"
+                    )
                     result = self._elementify(instance, data)
-                request.setHeader(
-                    b"content-type",
-                    (
-                        "text/{format}; charset=utf-8".format(
-                            format=str
-                        ).encode("charmap")
-                    ),
-                )
                 returnValue(result)
```

Comment:
do you think fixing `returnValue(x)` -> `return x` is in scope for this PR?

Reply:
That would mean not using `inlineCallbacks`… yeah.  It would be good to get rid of those everywhere though. I'll file a ticket.

Reply:
#410

Reply:
You can use regular return with inlineCallbacks

Reply:
Huh.  Been a long time.  Still, let's save that for later.